frappe/model/sync.py

```diff
@@ -135,3 +136,92 @@ def get_doc_files(files, start_path):
 							files.append(doc_path)
 
 	return files
+
+
+def remove_orphan_doctypes():
+	"""Find and remove any orpahned doctypes.
+
+	These are doctypes for which code and schema file is
+	delted but entry is present in DocType table.
+
+	Note: Deleting the entry doesn't delete any data.
+	So this is supposed to be non-destrictive operation.
+	"""
+
+	doctype_names = frappe.get_all("DocType", {"istable": 0, "custom": 0}, pluck="name")
```

Comment:
```suggestion
	doctype_names = frappe.get_all("DocType", {"istable": 0, "custom": 0}, pluck="name")
```

Any reason why child tables are ignored?

Reply:
- If we delete child tables, there is possibility where these tables are added as `table` field in some other DOCTYPE.

Reply:
Without controller files these are useless anyway. They will give `ImportError`.

Reply:
True. We can still delete them, doing so will throw an error that `doctype has been deleted.`